### PR TITLE
Draft documentation framework for data sources

### PR DESCRIPTION
**Pull request by aesharpe**
In the most recent changes I moved all of the data source files to a directory called `data_sources` in the `docs` directory. These data pages are accessible via index in the `data_sources.rst` file. 

I also went through and updated all the references to the data sources in the old `data_catalog.rst` file. Now, they should link to the new page.

### CODE REVIEW

**Comment by aesharpe** [comment]
In the most recent changes I moved all of the data source files to a directory called `data_sources` in the `docs` directory. These data pages are accessible via index in the `data_sources.rst` file. 

I also went through and updated all the references to the data sources in the old `data_catalog.rst` file. Now, they should link to the new page.

File: docs/data_sources/data_catalog.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+.. contents::
+
+-------------------------------------------------------------------------------
+Available Data
+-------------------------------------------------------------------------------
+
+.. _data-epaipm:
+
+EPA IPM
```
Comment (cmgosnell) [inline docs/data_sources/data_catalog.rst:15]: we are not really supporting IPM right now so I think we can remove this.

File: docs/data_sources/data_catalog.rst [hunk 2]
```diff
@@ -0,0 +1,126 @@
+.. _data-catalog:
```
Comment (cmgosnell) [inline docs/data_sources/data_catalog.rst:1]: perhaps we can squish this page into top-level Data Sources page?

so it'd just be a list of links to the available data and these shorter descriptions of Work in Progress and Future Data?

Comment (cmgosnell) [reply docs/data_sources/data_catalog.rst:1]: or this data-catalog page can be converted into a Future data/WIP page?? that might be cleaner

File: docs/eia860.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+.. todo::
+    Say more about legal background.
+
+The Form EIA-860 collects environmental equipment and generator-level data from
+extant as well as planned generation entities with one or more megawatts of
+capacity. Beginning in 2013, EIA-860 also collects construction cost data.
+
+Notable Irregularities
+^^^^^^^^^^^^^^^^^^^^^^
+
```
[the remark this answers is not in the document]
Comment (aesharpe) [reply docs/eia860.rst]: That's a good idea -- we should add that to the google doc version of this for now. The contents here are just a stand-in for when we finalize the contents in google docs.

File: docs/data_sources/data_catalog.rst (hunk 1 above)
Comment (aesharpe) [reply docs/data_sources/data_catalog.rst:15]: Should I also remove all references to the IPM in the documentation?

File: docs/data_sources/data_catalog.rst (hunk 2 above)
Comment (aesharpe) [reply docs/data_sources/data_catalog.rst:1]: ok I like the idea of Future/WIP for now!

File: docs/data_sources/data_catalog.rst (hunk 1 above)
Comment (aesharpe) [reply docs/data_sources/data_catalog.rst:15]: done

**Pull request merged**